src/api.rs

```diff
@@ -169,6 +169,14 @@ impl<L: Language> SyntaxNode<L> {
         self.raw.green()
     }
 
+    pub fn ptr(&self) -> Option<SyntaxPtr> {
+        self.raw.ptr()
+    }
+
+    pub fn resolve_from_here(&self, ptr: &SyntaxPtr) -> Option<SyntaxElement<L>> {
```

Comment:
I think this should `.unwrap` internally -- failure to resolve a ptr is always a programmer's error.